llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1462,15 +1462,22 @@ static void transformRecipestoEVLRecipes(VPlan &Plan, VPValue &EVL) {
         assert(OrigMask && "Unmasked recipe when folding tail");
         return HeaderMask == OrigMask ? nullptr : OrigMask;
       };
+      auto SetEVLForReversePointer = [&EVL](VPValue *V) -> void {
+        if (auto *R = dyn_cast_if_present<VPReverseVectorPointerRecipe>(
+                V->getDefiningRecipe()))
+          R->setOperand(1, &EVL);
+      };
 
       VPRecipeBase *NewRecipe =
           TypeSwitch<VPRecipeBase *, VPRecipeBase *>(CurRecipe)
               .Case<VPWidenLoadRecipe>([&](VPWidenLoadRecipe *L) {
                 VPValue *NewMask = GetNewMask(L->getMask());
+                SetEVLForReversePointer(L->getAddr());
```

Comment:
Would it be better to set EVL for all `VPReverseVectorPointerRecipe`? There could be other memory recipes in the future , e.g. interleave groups

Reply:
Updated, instead I traverse each user of VF to find all `VPReverseVectorPointerRecipe`.